elfcore/src/coredump.rs

```diff
@@ -521,16 +517,29 @@ fn get_va_regions(
                 // as it might be quite huge and may contains secrets, filter out by default.
                 // TODO: make optional.
 
-                // SAFETY: reading other process memory. The process is paused, the address
-                // comes from the kernel. The structure is repr(C) with no padding bytes,
-                // so all byte patterns are valid.
-                let maybe_elf_hdr = unsafe {
-                    let mut elf_hdr: Elf64_Ehdr = std::mem::zeroed();
-                    let elf_hdr_slice = slice::from_raw_parts_mut(
-                        &mut elf_hdr as *mut _ as *mut u8,
-                        std::mem::size_of::<Elf64_Ehdr>(),
-                    );
-                    match mem.read_exact_at(elf_hdr_slice, begin) {
+                let maybe_elf_hdr = {
+                    // SAFETY: while the all-zero byte content does not represent
+                    // a semantically valid value, the code below checks for the
+                    // correctness.
+                    let mut elf_hdr: Elf64_Ehdr = unsafe { std::mem::zeroed() };
```

Comment:
i wonder could we replace this with FromBytes::zeroed?

Reply:
That's possible at the cost of moving `Elf64_Ehdr` declaration into `elf.rs` (or could use a newtype but looked not very pretty) to be able to derive `FromBytes`. Probably that's a fair price for removing two unsafe blocks, the code looks less boilerplate.

While was in the area, removed `packed` from the structs due to `AsBytes` enforcing the absence of padding space (which is the intent) and placing no restrictions on alignment.